app/src/main/java/com/example/shareyourtrip/PostAdapter.java

```diff
@@ -46,17 +56,54 @@ public MyViewHolder(View view) {
                 description = (TextView) view.findViewById(R.id.description);
                 user = (TextView) view.findViewById(R.id.user);
                 date = (TextView) view.findViewById(R.id.date);
+
                 favButton = (ToggleButton) view.findViewById(R.id.fav_button);
                 likeButton = (ToggleButton) view.findViewById(R.id.like_button);
                 likeCount = (TextView) view.findViewById(R.id.like_count);
                 dislikeButton = (ToggleButton) view.findViewById(R.id.dislike_button);
                 dislikecount = (TextView) view.findViewById(R.id.dislike_count);
+
+
+                // on item click
+//                view.setOnClickListener(new View.OnClickListener() {
+//                    @Override
+//                    public void onClick(View v) {
+//                        // get position
+//                        int pos = getAdapterPosition();
+//
+//                        // check if item still exists
+//                        if (pos != RecyclerView.NO_POSITION) {
+//                            Post post = postList.get(pos);
+//                            clickedPost = new Post(post);
+//                        }
+//                    }
+//                });
+
+//                favButton.setOnClickListener(new View.OnClickListener() {
+//                    @Override
+//                    public void onClick(View v) {
+//                        int pos = getAdapterPosition();
+//
+//                        // check if item still exists
+//                        if (pos != RecyclerView.NO_POSITION) {
+//                            Post post = postList.get(pos);
+//                            clickedPost = new Post(post);
+//
+//                            postDAO.insertFavPost(post);
+//                        }
+//                    }
+//                });
             }
         }
 
+
         //Copy constructor
-        public PostAdapter(List<Post> postList) {
+        public PostAdapter(List<Post> postList, Context context) {
+
             this.postList = postList;
+            postDAO = new PostDAO(context);
+            currentContext = context;
+            favPostList = new ArrayList<Post>();
```

Comment:
A list of the user's favorite posts should be passed into this constructor as the postList parameter in the FavoriteActivity.